woocommerce/woocommerce-android
# [Add new coupon] Coupon type picker

Closes: #9470

This PR adds coupon type picker bottom sheet. It's shown when clicking the "+"  create coupon button in the coupon management screen.

### Description

The coupon type picker is needed to pass `Coupon.Type` value to the EditCoupon screen when creating a new coupon. 
Figma designs: O8MT8FHx8N2zP9pc2namgl-fi-1441%3A13027

### Testing instructions

1. Go to coupon management
2. Click "+"
3. Verify coupon type picker is shown and click on one of the types
4. Verify the correct coupon type is passed to the Edit Coupon (coupon creation) screen.

### Visual changes
<img width=300 src=https://github.com/woocommerce/woocommerce-android/assets/4527432/854238c7-4e2c-4bd0-89de-f8a283c49151/>
